azkaban-exec-server/src/main/java/azkaban/execapp/ExecutorConfigValidator.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2018 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.execapp;
+
+import azkaban.ConfigValidator;
+import azkaban.Constants.ConfigurationKeys;
+import azkaban.Constants.DefaultValue;
+import azkaban.utils.Props;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+/**
+ * Validator of Azkaban executor's configuration
+ */
+
+@Singleton
+public class ExecutorConfigValidator extends ConfigValidator {
```

Comment:
Is it better to use composition instead of inheritance here? Using inheritance for code reuse only is an anti-pattern IMHO.

Reply:
i think here is not just for code reuse? Semantically inheriting an abstract class makes sense, users are free to pass in customized logic to build their own validator right?

 Another example is rocksdb's CompactionFilter, which the db provides an interface, user can write their own compaction logic: https://github.com/facebook/rocksdb/blob/c364eb42b5fde3bf501e04c5a23f32c99ec4465c/utilities/blob_db/blob_compaction_filter.cc#L17